### PR TITLE
[Top nav] Use custom title bar buttons on windows & linux

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None (feature flagged)

**Description**
Fixes FG-1293

Use custom window controls on Linux, and [window controls overlay](https://www.electronjs.org/docs/latest/tutorial/window-customization#window-controls-overlay-macos-windows) on Windows.

image

### CODE REVIEW

File: packages/studio-base/src/components/AppBar/index.tsx [hunk 1]
```diff
@@ -164,66 +200,103 @@ export function AppBar(props: AppBarProps): JSX.Element {
...
+                      aria-haspopup="true"
+                      aria-expanded={userMenuOpen ? "true" : undefined}
+                      onClick={handleUserMenuClick}
+                      size="small"
+                    />
+                  ) : (
+                    <Button
+                      variant="contained"
+                      color="primary"
+                      onClick={() => {
```
Comment (defunctzombie) [inline packages/studio-base/src/components/AppBar/index.tsx:251]: I've found myself thinking we should push more on keeping inline functions beyond a single line out of the react/jsx part of the render code. I think it makes it easier to read and maintain the jsx markup

File: packages/studio-base/src/components/AppBar/index.tsx [hunk 2]
```diff
@@ -164,66 +200,103 @@ export function AppBar(props: AppBarProps): JSX.Element {
...
+                            cta: "sign-in",
+                          });
+                        }
+                      }}
+                    >
+                      Sign in
+                    </Button>
+                  ))}
+                {showCustomWindowControls && (
+                  <>
```
Comment (defunctzombie) [inline packages/studio-base/src/components/AppBar/index.tsx:265]: Possibly worth turning these controls into their own component and using that within the app bar?

**Comment by amacneil** [comment]
How do other electron apps on linux+windows handle the icon problem? Is there no way to get more native looking icons?

Haven't tested myself, but the screenshot looks nice. Low risk since it is behind a feature flag.

**Comment by jtbandes** [comment]
We could try this controls overlay thing instead: https://www.electronjs.org/docs/latest/tutorial/window-customization#window-controls-overlay-macos-windows
But that’s not supported on Linux. Anyway aren’t there multiple Linux window managers that would have different-looking buttons anyway? Do we want to try and detect the window manager and use custom icons that match more closely?

**Review by defunctzombie — APPROVED** [review]

**Pull request merged**